src/brevitas_examples/llm/llm_quant/gpxq.py

```diff
@@ -149,24 +150,22 @@ def apply_gptq(
                 gptq.update()
 
 
-@torch.no_grad()
-def apply_gpfq(
+def _gpfq_callback(
         model,
         dataloader,
         act_order=True,
         group_of_parallel_layers=None,
         block_name=None,
         max_accumulator_bit_width=None,
-        max_accumulator_tile_size=None):
+        max_accumulator_tile_size=None,
+        gpfq_class=GPFQ):
```

Comment:
Not a massive fan of this, since Qronos and GPFQ are related but not the same thing.
The idea is correct, maybe it will require a change of names. Something like `single_forward_optimization` vs `double_forward_optimization`.

If you don't want to change this now, I guess it can wait.